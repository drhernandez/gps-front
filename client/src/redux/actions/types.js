const TOGGLE_SIDEBAR = 'TOGGLE_SIDEBAR';
const SET_USER_INFO = 'SET_USER_INFO';
const REMOVE_USER_INFO = 'REMOVE_USER_INFO';

export {
  TOGGLE_SIDEBAR,
  SET_USER_INFO,
  REMOVE_USER_INFO
};