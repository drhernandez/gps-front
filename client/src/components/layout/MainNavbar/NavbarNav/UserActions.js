import React from "react";
import { Link } from "react-router-dom";
import store from "../../../../redux/store";
import {
  Dropdown,
  DropdownToggle,
  DropdownMenu,
  DropdownItem,
  Collapse,
  NavItem,
  NavLink
} from "shards-react";

import AuthService from "../../../../api/services/authService";


export default class UserActions extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      visible: false,
      userInfo: undefined
    };

    this.toggleUserActions = this.toggleUserActions.bind(this);
  }

  componentDidMount() {
    const state = store.getState();
    this.setState({
      userInfo: state.userInfo
    })
  }

  toggleUserActions() {
    this.setState({
      visible: !this.state.visible
    });
  }

  logout() {
    AuthService.logout();
  }

  render() {
    return (
      <NavItem tag={Dropdown} caret toggle={this.toggleUserActions}>
        <DropdownToggle caret tag={NavLink} className="d-flex align-items-center h-100 text-nowrap px-3">
          <span className="d-none d-inline-block">{this.state.userInfo && this.state.userInfo.userName + " " + this.state.userInfo.userLastName}</span>
        </DropdownToggle>
        <Collapse tag={DropdownMenu} right small open={this.state.visible}>
          <DropdownItem tag={Link} to="user-profile">
            <i className="material-icons">&#xE7FD;</i> Profile
          </DropdownItem>
          <DropdownItem tag={Link} to="edit-user-profile">
            <i className="material-icons">&#xE8B8;</i> Edit Profile
          </DropdownItem>
          <DropdownItem divider />
          <DropdownItem tag={Link} onClick={() => this.logout()} to="/" className="text-danger">
            <i className="material-icons text-danger">&#xE879;</i> Logout
          </DropdownItem>
        </Collapse>
      </NavItem>
    );
  }
}
