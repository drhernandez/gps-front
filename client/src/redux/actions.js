const TOGGLE_SIDEBAR = 'TOGGLE_SIDEBAR';

function toogleSidebarAction() {
  return {
    type: TOGGLE_SIDEBAR
  }
}

const types = {
  TOGGLE_SIDEBAR
};

const actions = {
  toogleSidebarAction
}

export { types, actions };