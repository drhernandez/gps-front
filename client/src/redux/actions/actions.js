import { TOGGLE_SIDEBAR } from './types';

function toogleSidebarAction() {
  return {
    type: TOGGLE_SIDEBAR
  }
}

export {
  toogleSidebarAction
};