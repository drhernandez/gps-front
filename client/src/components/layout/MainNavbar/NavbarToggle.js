import React from "react";
import { connect } from 'react-redux'
import { toogleSidebarAction } from "../../../redux/actions/actions";

class NavbarToggle extends React.Component {

  render() {
    return (
      <nav className="nav">
        {/* eslint-disable-next-line */}
        <a href="#" onClick={this.props.toogleSidebarAction} className="nav-link nav-link-icon toggle-sidebar d-sm-inline d-md-inline d-lg-none text-center">
          <i className="material-icons">&#xE5D2;</i>
        </a>
      </nav>
    )
  }
}

function mapDispatchToProps(dispatch) {
  return {
    toogleSidebarAction: () => dispatch(toogleSidebarAction())
  }
}

export default connect(null, mapDispatchToProps)(NavbarToggle);
