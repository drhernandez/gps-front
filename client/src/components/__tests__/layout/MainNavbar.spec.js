import React from 'react'
import renderer from 'react-test-renderer'
import { mount } from 'enzyme'
import { Router } from "react-router-dom"
import { createMemoryHistory } from 'history'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'
const mockStore = configureStore([])

import { toogleSidebarAction } from "../../../redux/actions/actions"
import MainNavbar from '../../layout/MainNavbar/MainNavbar'
import NavbarToggle from '../../layout/MainNavbar/NavbarToggle'
import NavbarNav from '../../layout/MainNavbar/NavbarNav/NavbarNav'
import UserActions from '../../layout/MainNavbar/NavbarNav/UserActions'

let history
let store
describe('MainNavbar components', () => {
  beforeEach(() => {
    jest.resetModules()
    history = createMemoryHistory()
    store = mockStore({
      userInfo: {
        id: 1,
        name: 'test',
        lastName: 'test'
      },
    })
    store.dispatch = jest.fn()
  })

  it('MainNavbar snapshot', () => {
    const component = renderer.create(
      <Router history={history}>
        <Provider store={store}>
          <MainNavbar />
        </Provider>
      </Router>
    )

    expect(component.toJSON()).toMatchSnapshot()
  })

  it('NavbarToggle snapshot', () => {
    const component = renderer.create(
      <Router history={history}>
        <Provider store={store}>
          <NavbarToggle />
        </Provider>
      </Router>
    )

    expect(component.toJSON()).toMatchSnapshot()
  })

  it('NavbarToggle should dispatch toogleSidebarAction', () => {
    const wrapper = mount(
      <Router history={history}>
        <Provider store={store}>
          <NavbarToggle />
        </Provider>
      </Router>
    )

    wrapper.find('a').simulate('click')

    expect(store.dispatch).toHaveBeenCalledTimes(1);
    expect(store.dispatch).toHaveBeenCalledWith(toogleSidebarAction());
  })

  it('NavbarNav snapshot', () => {
    const component = renderer.create(
      <Router history={history}>
        <Provider store={store}>
          <NavbarNav />
        </Provider>
      </Router>
    )

    expect(component.toJSON()).toMatchSnapshot()
  })

  it('UserActions snapshot', () => {
    const component = renderer.create(
      <Router history={history}>
        <Provider store={store}>
          <UserActions />
        </Provider>
      </Router>
    )

    expect(component.toJSON()).toMatchSnapshot()
  })
})