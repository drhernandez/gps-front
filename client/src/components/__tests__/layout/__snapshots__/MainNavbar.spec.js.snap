// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainNavbar components MainNavbar snapshot 1`] = `
<div
  className="main-navbar bg-white sticky-top"
>
  <div
    className="p-0 container"
    style={
      Object {
        "maxWidth": "100%",
      }
    }
  >
    <nav
      className="d-flex flex-row-reverse  p-0 navbar navbar-light"
    >
      <ul
        className="border-left flex-row navbar-nav"
      >
        <div
          className="nav-item dropdown"
        >
          <a
            aria-haspopup={true}
            className="d-flex align-items-center h-100 text-nowrap px-3 dropdown-toggle nav-link"
            onClick={[Function]}
          >
            <span
              className="d-none d-inline-block"
            >
              test test
            </span>
          </a>
          <div
            className="collapse dropdown-menu dropdown-menu-small dropdown-menu-right"
            role="menu"
            style={
              Object {
                "display": false,
                "height": null,
              }
            }
            tabIndex="-1"
          >
            <a
              className="dropdown-item"
              href="/user-profile"
              onClick={[Function]}
              tabIndex="0"
            >
              <i
                className="material-icons"
              >
                
              </i>
               Profile
            </a>
            <a
              className="dropdown-item"
              href="/edit-user-profile"
              onClick={[Function]}
              tabIndex="0"
            >
              <i
                className="material-icons"
              >
                
              </i>
               Edit Profile
            </a>
            <div
              className="dropdown-divider"
              onClick={[Function]}
              tabIndex="-1"
            />
            <a
              className="text-danger dropdown-item"
              href="/logout"
              onClick={[Function]}
              tabIndex="0"
            >
              <i
                className="material-icons text-danger"
              >
                
              </i>
               Logout
            </a>
          </div>
        </div>
      </ul>
      <nav
        className="nav"
      >
        <a
          className="nav-link nav-link-icon toggle-sidebar d-sm-inline d-md-inline d-lg-none text-center"
          href="#"
          onClick={[Function]}
        >
          <i
            className="material-icons"
          >
            
          </i>
        </a>
      </nav>
    </nav>
  </div>
</div>
`;

exports[`MainNavbar components NavbarNav snapshot 1`] = `
<ul
  className="border-left flex-row navbar-nav"
>
  <div
    className="nav-item dropdown"
  >
    <a
      aria-haspopup={true}
      className="d-flex align-items-center h-100 text-nowrap px-3 dropdown-toggle nav-link"
      onClick={[Function]}
    >
      <span
        className="d-none d-inline-block"
      >
        test test
      </span>
    </a>
    <div
      className="collapse dropdown-menu dropdown-menu-small dropdown-menu-right"
      role="menu"
      style={
        Object {
          "display": false,
          "height": null,
        }
      }
      tabIndex="-1"
    >
      <a
        className="dropdown-item"
        href="/user-profile"
        onClick={[Function]}
        tabIndex="0"
      >
        <i
          className="material-icons"
        >
          
        </i>
         Profile
      </a>
      <a
        className="dropdown-item"
        href="/edit-user-profile"
        onClick={[Function]}
        tabIndex="0"
      >
        <i
          className="material-icons"
        >
          
        </i>
         Edit Profile
      </a>
      <div
        className="dropdown-divider"
        onClick={[Function]}
        tabIndex="-1"
      />
      <a
        className="text-danger dropdown-item"
        href="/logout"
        onClick={[Function]}
        tabIndex="0"
      >
        <i
          className="material-icons text-danger"
        >
          
        </i>
         Logout
      </a>
    </div>
  </div>
</ul>
`;

exports[`MainNavbar components NavbarToggle snapshot 1`] = `
<nav
  className="nav"
>
  <a
    className="nav-link nav-link-icon toggle-sidebar d-sm-inline d-md-inline d-lg-none text-center"
    href="#"
    onClick={[Function]}
  >
    <i
      className="material-icons"
    >
      
    </i>
  </a>
</nav>
`;

exports[`MainNavbar components UserActions snapshot 1`] = `
<div
  className="nav-item dropdown"
>
  <a
    aria-haspopup={true}
    className="d-flex align-items-center h-100 text-nowrap px-3 dropdown-toggle nav-link"
    onClick={[Function]}
  >
    <span
      className="d-none d-inline-block"
    >
      test test
    </span>
  </a>
  <div
    className="collapse dropdown-menu dropdown-menu-small dropdown-menu-right"
    role="menu"
    style={
      Object {
        "display": false,
        "height": null,
      }
    }
    tabIndex="-1"
  >
    <a
      className="dropdown-item"
      href="/user-profile"
      onClick={[Function]}
      tabIndex="0"
    >
      <i
        className="material-icons"
      >
        
      </i>
       Profile
    </a>
    <a
      className="dropdown-item"
      href="/edit-user-profile"
      onClick={[Function]}
      tabIndex="0"
    >
      <i
        className="material-icons"
      >
        
      </i>
       Edit Profile
    </a>
    <div
      className="dropdown-divider"
      onClick={[Function]}
      tabIndex="-1"
    />
    <a
      className="text-danger dropdown-item"
      href="/logout"
      onClick={[Function]}
      tabIndex="0"
    >
      <i
        className="material-icons text-danger"
      >
        
      </i>
       Logout
    </a>
  </div>
</div>
`;
